### ISSUE TITLE
ansible_mitogen prints a traceback if interrupted using Ctrl-C

### ISSUE THREAD

**Issue by willmerae** [dw/mitogen#241]
Interrupting a playbook run with vanilla Ansible looks like

```
TASK [Gathering Facts] **********************************************************************************
Wednesday 02 May 2018  15:25:35 +0100 (0.23s)       0:00:00.768 *************** 
^C [ERROR]: User interrupted execution
```
 if Mitogen (76102927c81d5dfcc6c91565b2984f24dd3ee678) is enabled then I get 
```
TASK [redacted : lorum ipsom ] **************************************************************************
^C [ERROR]: User interrupted execution

 [ERROR]: User interrupted execution

ERROR! [pid 8087] 15:20:17.991597 E mitogen.ctx.local.9275: mitogen: ExternalContext.main() crashed
Traceback (most recent call last):
  File "<stdin>", line 1684, in main
  File "<stdin>", line 257, in <lambda>
  File "<stdin>", line 1652, in _dispatch_calls
  File "<stdin>", line 460, in __iter__
  File "<stdin>", line 449, in get
  File "<stdin>", line 1043, in get
  File "<stdin>", line 1050, in _get_sleep
  File "<stdin>", line 207, in io_op
KeyboardInterrupt
ERROR! [pid 8087] 15:20:18.032481 E mitogen.ctx.local.13691: mitogen: ExternalContext.main() crashed
Traceback (most recent call last):
  File "<stdin>", line 1684, in main
  File "<stdin>", line 257, in <lambda>
  File "<stdin>", line 1652, in _dispatch_calls
  File "<stdin>", line 460, in __iter__
  File "<stdin>", line 449, in get
  File "<stdin>", line 1043, in get
  File "<stdin>", line 1050, in _get_sleep
  File "<stdin>", line 207, in io_op
KeyboardInterrupt
```
This is cosmetic thing only. I just wanted to make a note of it.

**Comment by dw** [comment]
There are quite a few of these... there is a race during shutdown where you
can also see "no route" messages. I think something about shutdown might be
broken just now (although I did check and everything definitely does seem
to be dying reliably with CTRL+C)

**Comment by willmerae** [comment]
I assumed Ansible contains a "try: run_playbook(); except KeyboardInterrupt: pass" somewhere that Mitogen happens to bypass.

**Comment by dw** [comment]
The multiplexer no longer lives in the top-level process, it's forked off early because Ansible forks so much that the broker thread reliably deadlocks, because of other code taking locks during fork. So shutdown actually runs asynchronous to Ansible's own shutdown for quite some time.

In this case the exception is due to ExternalContext.main() not catching KeyboardInterrupt, which it is usually isolated from, and usually we want to log this condition as an error (unexpected signal), but in Ansible's case due to the haste in which the connection multiplexer was added, I never isolated it from signals originating on the terminal running Ansible.

Going to keep this bug open as the fix I just checked in is incomplete - we need to add setsid() to the multiplexer process, after ensuring it won't break anything.

btw there is an incomplete doc diagram covering this process layout :)
